src/Ledger/TokenAlgebra/ValueRelation.lagda

```diff
@@ -0,0 +1,471 @@
+open import Data.Product.Base            using (swap)
+open import Data.Sum.Base                using ([_,_]′)
+
+open import Ledger.Prelude               hiding (Coin ; Rel ; _>_)
+open import Ledger.TokenAlgebra gc PolicyID
+                                         using (AssetID ; AssetName ; Quantity ; TokenAlgebraRel)
+open import Relation.Binary              using (REL ; Rel ; _⇒_ ; IsEquivalence ; Decidable)
+                                         renaming (_⇔_ to _⇐⇒_ )
+open import Relation.Binary.Definitions  using (Decidable ; DecidableEquality)
+
+open import Relation.Binary.PropositionalEquality.Core using ( module ≡-Reasoning )
+
+
+private
+  variable
+    A B : Type
+    C : REL A B 0ℓ → Type 0ℓ
+
+open ≡-Reasoning
+
+\end{code}
+
+\subsection{The identity of the value monoid}
+
+We'll use \AgdaBound{ι} to denote the identity for the Value monoid.
+
+\begin{code}
+ι : REL AssetID Quantity 0ℓ
+ι _ q = q ≡ 0
+
+ι-left-unique : left-unique-rel ι
+ι-left-unique 0b 0b' = trans 0b (sym 0b')
+
+ι-left : REL AssetID Quantity 0ℓ → REL AssetID Quantity 0ℓ
+ι-left R aid q = ∃[ qι ] (∃[ qu ] ι aid qι × R aid qu × q ≡ qι + qu)
+
+ι-right : REL AssetID Quantity 0ℓ → REL AssetID Quantity 0ℓ
+ι-right R aid q = ∃[ qu ] (∃[ qι ] R aid qu × ι aid qι × q ≡ qu + qι)
+
+ι-left-lemma1 : ∀{aid q}(R : REL AssetID Quantity 0ℓ)(h : ι-left R aid q) → proj₁ h ≡ 0
+ι-left-lemma1 _ h = (proj₁ (proj₂ (proj₂ h)))
+
+ι-left-lemma2 : ∀{aid q}(R : REL AssetID Quantity 0ℓ)(h : ι-left R aid q) → q ≡ proj₁ (proj₂ h)
+ι-left-lemma2 {aid}{q} R h = let qι = proj₁ h ; qu = proj₁ (proj₂ h) in
+ begin
+   q        ≡⟨ proj₂ (proj₂ (proj₂ (proj₂ h))) ⟩
+   qι + qu  ≡⟨ cong (λ x → x + qu) (ι-left-lemma1 R h) ⟩
+   0 + qu   ≡⟨ +-identityˡ qu ⟩
+   qu       ∎
+
+ι-right-lemma1 : ∀{aid q}(R : REL AssetID Quantity 0ℓ)(h : ι-right R aid q) → proj₁ (proj₂ h) ≡ 0
+ι-right-lemma1 _ h = proj₁ (proj₂ (proj₂ (proj₂ h)))
+
+ι-right-lemma2 : ∀{aid q}(R : REL AssetID Quantity 0ℓ)(h : ι-right R aid q) → q ≡ proj₁ h
+ι-right-lemma2 {aid}{q} R h = let qu = proj₁ h ; qι = proj₁ (proj₂ h) in
+ begin
+   q        ≡⟨ proj₂ (proj₂ (proj₂ (proj₂ h))) ⟩
+   qu + qι  ≡⟨ cong (λ x → qu + x) (ι-right-lemma1 R h) ⟩
+   qu + 0   ≡⟨ +-identityʳ qu ⟩
+   qu       ∎
+
+ι-left⇒R : ∀{R} → ι-left R ⇒ R
+ι-left⇒R {R} {aid} {q} h = Goal
+ where
+ qu≡q : proj₁ (proj₂ h) ≡ q
+ qu≡q = sym (ι-left-lemma2 R h)
+
+ Goal : R aid q
+ Goal = subst (λ x → R aid x) qu≡q (proj₁ (proj₂ (proj₂ (proj₂ h))))
+
+R⇒ι-left : ∀{R} → R ⇒ ι-left R
+R⇒ι-left {R} {aid} {q} Rh = 0 , q , refl , Rh , sym (+-identityˡ q)
+
+ι-right⇒R : ∀{R} → ι-right R ⇒ R
+ι-right⇒R {R} {aid} {q} h = subst (λ x → R aid x) qu≡q (proj₁ (proj₂ (proj₂ h)))
+ where
+ qu≡q : proj₁ h ≡ q
+ qu≡q = sym (ι-right-lemma2 R h)
+
+R⇒ι-right : ∀{R} → R ⇒ ι-right R
+R⇒ι-right {R} {aid} {q} Rh = q , 0 , Rh , refl , sym (+-identityʳ q)
+
+\end{code}
+
+\subsection{Equivalence on the value monoid}
+
+\begin{code}
+-- the following probably belongs somewhere else
```

Comment:
Are you sure this isn't in the stdlib yet? If not, I'd move it to `Relation.Binary.Properties.Ext`. I'm using `Ext` to denote modules that contain things that should have been in the stdlib (and would like to merge upstream some time in the future).